docs/markdown/DevGuide_PythonProcessors.md

```diff
@@ -0,0 +1,82 @@
+
+## Processor Definition
+First of all `inviwopy` needs to be imported (called `ivw` in the following) to get access to `ivw.Processor`, `ivw.properties.*`, `ivw.ports.*` and `ivw.glm.*` wrappers. In our NumpySource example the imports are:
+```python
+import inviwopy as ivw
+from inviwopy.properties import FileProperty, InvalidationLevel
+from inviwopy.data       import Volume, VolumeOutport
+
+import numpy as np
+from pathlib import Path
+```
+
+Using those wrappers you can define Inviwo processors very similar to how it is done in C++. The actual processor definition happens inside your own processors class, inheriting from `ivw.Processor` and defining all the following methods:
+```python
+class NumpySource(ivw.Processor):
+    def __init__(self, id, name):    # Default processor signature
+        super().__init__(id, name)   # Call super class (Processor) with id, name
+
+        [ Here comes normal constructor contents like adding ports, properties ]
+
+    @staticmethod
+    def processorInfo():                 # General information on this processor
+        return ivw.ProcessorInfo(
+            classIdentifier = "org.inviwo.numpysource",
+    		displayName = "NumpySource",
+    		category = "Python",
+    		codeState = ivw.CodeState.Stable,
+    		tags = ivw.Tags.PY
+        )
+
+    def getProcessorInfo(self): return NumpySource.processorInfo()
+
+    def initializeResources(self):
+        pass
+
+    def process(self):
+        pass
+```
+
+As you can see, just as in C++, a processor needs to define a constructor to define all ports, properties etc., some information about the processor itself, the `initializeResources()` and the `process()` method. If you don't need `initializeResources()`, just define it with the `pass` no-op as function body.
+Also do not forget the call to `ivw.Processor.__init__` in your processors `__init__`.
+
+In our `NumpySource` example we can use the following `__init__`:
+```python
+class NumpySource(ivw.Processor):
+    def __init__(self, id, name):
+        ivw.Processor.__init__(self, id, name)
+        self.outport = VolumeOutport("outport")   # Define Outport
+        self.addOutport(self.outport)            # Add port to processor
+
+        self.file = FileProperty("file", "Numpy Volume (.npy)",
+            invalidationLevel=InvalidationLevel(2)) # invalidate resources on change
+        self.addProperty(self.file)
+        self.array = None    # Init to None
+```
+This `__init__` defines a volume outport to pass the loaded array to the network and a file property to locate the serialized Numpy array. Note that this `FileProperty` has its `invalidationLevel=InvalidationLevel(2)`, which lets the `FileProperty` invalidate the processors resources upon changing the property. This will automatically call the `invalidateResources()` method which will take care of actually loading the Numpy file (see below).
```

Comment:
invalidateResources  -> initializeResources

Reply:
In principle you can just do the file load in the process. It would not make much difference here...

Reply:
Isn't this separation why `initializeResources()` exists in the first place? I agree that for this example it does not really matter, but I thought I would rather include this "convention" here to make sure that new users can adopt it, even if the example could be easier the other way.

Reply:
Actually I'm not 100% sure on why we have this, you might have to ask Timo :) 
But the main reason I think is if your processor depends on some resources that you don't want to reload on every process call but rather only at the beginning. The typical case in inviwo being a compiled shader. Then you only run initializeResources when you change something that will requite the shader to recompile. 
The main difference here being that for the soruce processor, the only reason it has for ever processing is to load the data. And since there is no reason to call process but for loading the data. There is no need to do that in initializeResources. If on the other hand initializeResources loaded the data, and process then took that data and did some processing on that depending on some property. Then I think that would make more sense. 
In this case you have to store the data as a member variable in the processor just to pass it from initializeResources to process which I think just add unnecessary complexity in this case.